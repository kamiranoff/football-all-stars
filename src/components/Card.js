import React from 'react';
import { Text, View, Image, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { SCREEN_HEIGHT } from '../constants';

const styles = StyleSheet.create({
  container: { alignItems: 'center' },
  image: { height: SCREEN_HEIGHT / 2 },
  text: { backgroundColor: 'white', padding: 20, textAlign: 'center' }
});

const CardComponent = ({ item }) => (
  <View style={styles.container}>
    <Image source={item.uri} style={styles.image} />
    <Text style={styles.text}>
      {item.text}
    </Text>
  </View>
);

CardComponent.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.number.isRequred,
    text: PropTypes.string.isRequred,
    uri: PropTypes.number.isRequired,
  }).isRequired,
};

export default CardComponent;