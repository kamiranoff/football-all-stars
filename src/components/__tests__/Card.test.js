import React from 'react';
import Card from '../Card';

import renderer from 'react-test-renderer';

const item = {
  id: 1,
  text: 'hello',
  uri: 1,
}

it('renders without crashing', () => {
  const rendered = renderer.create(<Card item={item} />).toJSON();
  expect(rendered).toMatchSnapshot();
});
