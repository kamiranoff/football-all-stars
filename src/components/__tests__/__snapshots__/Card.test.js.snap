// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 667,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "white",
        "padding": 20,
        "textAlign": "center",
      }
    }
  >
    hello
  </Text>
</View>
`;
